### PR TITLE
Doc CDC backfill for shared CDC tables

### LINKED ISSUE #1825: Document: feat(cdc): support disable cdc backfill and only consumes from latest changelog
This issue tracks the documentation update needed for the merged PR #14718.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/14718
Source PR Merged At: 2024-02-01T05:38:28Z

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Document `snapshot` parameter for MySQL CDC tables created from shared source.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/14718

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1825

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by emile-00** [comment]
@StrikeW Based on the code PR I assumed the `snapshot` option only applies for tables created from a shared source. Please let me know if that's incorrect.

File: docs/guides/ingest-from-postgres-cdc.md
Comment (StrikeW) [inline docs/guides/ingest-from-postgres-cdc.md:1]: This part:
>Syntax for creating a CDC table. Note that a primary key is required.
 >```sql
 > CREATE {TABLE | SOURCE} [ IF NOT EXISTS ] table_name (
 >   column_name data_type PRIMARY KEY , ...
 >   PRIMARY KEY ( column_name, ... )
>  ) 
> WITH (
>    connector='postgres-cdc',
>    connector_parameter='value', ...
>  )
>  [ FORMAT DEBEZIUM ENCODE JSON ];
> ```

We should remove the `SOURCE` from this syntax. The `SOURCE` is only allowed in the shared cdc source syntax.

**Review by StrikeW — APPROVED** [review]
Rest lgtm

**Review by ShanlanLi — APPROVED** [review]
LGTM

**Review by WanYixian — APPROVED** [review]

**Pull request merged**